### PR TITLE
bpo-42116: Fix inspect.getsource handling of trailing comments

### PR DESCRIPTION
**Pull request by iritkatriel**
https://bugs.python.org/issue42116

### CODE REVIEW

**Comment by taleinat** [comment]
ISTM the we should only include comments indented at least as much as the function's inner indentation, rather than anything more than the outer indentation.

For example, in the following case, I would not expect the comment to be considered part of the function's code:

```python3
def foo():
    print("Hello")
  # a comment?
```

**Comment by iritkatriel** [comment]
> ISTM that we should only include comments indented at least as much as the function's inner indentation, rather than anything more than the outer indentation.

That's a good idea. It will make the code simpler, and we're talking about a non-PEP8 compliant edge case anyway.

File: Lib/test/inspect_fodder.py [hunk 1]
```diff
@@ -91,3 +91,24 @@ def as_method_of(self, obj):
...
+       # line 105
+    # after f
+
+    # before asyncf - line 108
+    async def asyncf(self):
+        # start asyncf
+        return 2
+        # end asyncf
+       # after asyncf - line 113
+  # after WhichComments - line 114
```
Comment (taleinat) [inline Lib/test/inspect_fodder.py]: Perhaps add a trailing comment for the class block which should be included?
```suggestion
    # end of WhichComments - line 114
  # after WhichComments - line 115
```

**Review by taleinat — APPROVED** [review]
Looks good! Just one minor suggestion for one of the test cases.

File: Lib/test/inspect_fodder.py (hunk 1 above)
Comment (iritkatriel) [reply Lib/test/inspect_fodder.py]: Good catch. I'll commit that (needs an update in the expected end line # in test_inspect as well).

**Pull request merged**